packages/block-library/src/quote/v2/transforms.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	createBlock,
+	parseWithAttributeSchema,
+	rawHandler,
+	serialize,
+} from '@wordpress/blocks';
+
+const transforms = {
+	from: [
+		{
+			type: 'block',
+			blocks: [ 'core/pullquote' ],
+			transform: ( { value, citation } ) => {
+				return createBlock(
+					'core/quote',
+					{
+						attribution: citation,
+					},
+					parseWithAttributeSchema( value, {
+						type: 'array',
+						source: 'query',
+						selector: 'p',
+						query: {
+							content: {
+								type: 'string',
+								source: 'text',
+							},
+						},
+					} ).map( ( { content } ) =>
+						createBlock( 'core/paragraph', { content } )
+					)
+				);
+			},
+		},
+		{
+			type: 'prefix',
+			prefix: '>',
+			transform: ( content ) =>
+				createBlock( 'core/quote', {}, [
+					createBlock( 'core/paragraph', { content } ),
+				] ),
+		},
+		{
+			type: 'raw',
+			schema: ( { phrasingContentSchema } ) => ( {
+				figure: {
+					require: [ 'blockquote' ],
+					children: {
+						blockquote: {
+							children: '*',
+						},
+						figcaption: {
+							children: phrasingContentSchema,
+						},
+					},
+				},
+			} ),
+			isMatch: ( node ) =>
+				node.nodeName === 'FIGURE' &&
+				!! node.querySelector( 'blockquote' ),
+			transform: ( node ) => {
+				const cite = node.querySelector( 'cite' );
+				const attribution =
+					cite?.innerHTML ??
+					node.querySelector( 'figcaption' )?.innerHTML;
+				cite?.parentNode?.removeChild( cite );
```

Comment:
I'm not sure if it's important to look for the cite in pasted content. It could be bad if there are multiple cites or if the cite is actually part of the content. We were using cite wrongly before, so I would omit cite handling here.

Reply:
This is only to provide backwards-compatibility with the existing `raw` behavior, that would handle this nicely:

```html
<p>content before</p>
<blockquote><p>First paragraph.</p><p>Second paragraph.</p><cite>Author</cite></blockquote>
<p>content after</p>
```

If we don't do this, things that worked before would no longer work.

Reply:
I think that's ok, we don't need back compatibility for paste. This markup would be very rare outside of WP. This action could be more destructive if it is NOT used as a caption. See https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-cite-element.

Reply:
OK, I'll go with your instinct. Reverted at https://github.com/WordPress/gutenberg/pull/39718/commits/9756835ec82bbaf41be12e9b13d1af7cb1c9f773